syntax = "proto3"; // Use Protocol Buffers version```proto3``` syntax

package greet; // Optional: Namespace for generated code

option java_package = "com.example.grpc";

service Greeter {
    rpc SayHello (HelloRequest) returns (HelloResponse) {}
    rpc SendImage (ImageRequest) returns (ImageResponse) {}
}

message HelloRequest {
    string name=1;
    string message=2;
}

message HelloResponse {
    string message=1;
}

message ImageRequest {
    bytes activations = 1;
    int32 height = 2;
    int32 width = 3;
    int32 channels = 4;
}

message ImageResponse {
    string prediction = 1;
}